### PR TITLE
test_remoteexecution.py::AnsibleREXTestCase::test_positive_run_job

### PR DESCRIPTION
**Pull request by jameerpathan111**
Test Result:
```
$ pytest tests/foreman/cli/test_remoteexecution.py::AnsibleREXTestCase -k test_positive_run_job 
========================================================================= test session starts ==========================================================================
collecting 9 items                                                                                                                                                     
collected 9 items / 8 deselected                                                                                                                                       
tests/foreman/cli/test_remoteexecution.py .                                                                                                                      [100%]

=============================================================== 1 passed, 8 deselected in 209.33 seconds ===============================================================
```

### CODE REVIEW

File: tests/foreman/cli/test_remoteexecution.py [hunk 1]
```diff
@@ -941,8 +942,26 @@ def setUpClass(cls):
...
+        cls.org = make_org()
+        cls.client = VirtualMachine(distro=DISTRO_RHEL7)
+        cls.client.create()
+        cls.client.install_katello_ca()
+        cls.client.register_contenthost(
+            org=cls.org['label'],
+            lce='Library'
+        )
+        cls.assertTrue(cls.client.subscribed, True)
+        Settings.set({'name': 'remote_execution_connect_by_ip',
```
Comment (rplevka) [inline tests/foreman/cli/test_remoteexecution.py]: This doesn't comply with the isolation - you are touching the global settings.
If there was another test relying on this setting, you would directly intefere with it.

- That is why the rest of the tests (with name *_by_ip) rather uses the host parameter. That way you only tell foreman to use ip for the very specific host unique to your test. 

```python
Host.set_parameter({
  'host': self.client.hostname,
  'name': 'remote_execution_connect_by_ip',
  'value': 'True',
})
```

Comment (jameerpathan111) [reply tests/foreman/cli/test_remoteexecution.py]: thanks @rplevka , I will only change the parameters for that specific host and won't interfere with global settings.

**Comment by jameerpathan111** [comment]
@rplevka I have done the changes you suggested.

Test  Result:
```
$ pytest tests/foreman/cli/test_remoteexecution.py::AnsibleREXTestCase -k test_positive_run_job
========================================================================= test session starts ==========================================================================
collecting 9 items                                                                                                                                                     
collected 9 items / 8 deselected                                                                                                                                       
tests/foreman/cli/test_remoteexecution.py .                                                                                                                      [100%]

=============================================================== 1 passed, 8 deselected in 193.42 seconds ===============================================================
```

**Comment by rplevka** [comment]
@pondrejk you've got experience wit hthe REX jobs, could you give this one a quick review, pls?

**Pull request merged**